### ISSUE TITLE
[R] Api Error

### ISSUE THREAD

**Issue by UpsideDownRide** [Microsoft/LightGBM#949]
When trying to run the model I get the following error.

## Environment info
Operating System: Windows Version	10.0.16294 Build 16294
CPU: i7-5820K
C++/Python/R version: R 3.4.2

## Error Message:

Error in lgb.call("LGBM_DatasetCreateFromMat_R", ret = handle, private$raw_data,  : 
  api error: Unknown parameter: showsd

traceback()

5: stop("api error: ", lgb.encode.char(err_msg, act_len))
4: lgb.call("LGBM_DatasetCreateFromMat_R", ret = handle, private$raw_data, 
       nrow(private$raw_data), ncol(private$raw_data), params_str, 
       ref_handle)
3: data$construct()
2: lgb.train(params, dtrain, nrounds, valids, verbose = verbose, 
       eval_freq = eval_freq, early_stopping_rounds = early_stopping_rounds, 
       init_model = init_model, callbacks = callbacks, ...)
1: lightgbm(data = as.matrix(train$data), label = train$label, num_leaves = 4, 
       learning_rate = 1, nrounds = 2, objective = "binary")

**Comment by huanzhang12** [comment]
It seems you add `showsd` to the parameter list `params` of `lgb.train`? `showsd` is not an acceptable parameter in this list; Instead, it is a valid parameter for the `lgb.train` function itself. So instead of doing

```
params <- list(showsd = True, ...)
lgb.train(params, ...)
```

you should do this:

```
lgb.train(params, showsd = True, ...)
```
